common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -554,6 +568,49 @@ def should_force_account_creation():
         # We know they did not just login, because the login process rejects unverified users.
 
 
+def create_user(strategy, details, user=None, autoprovision=False, *args, **kwargs):
+    """
+    Creates user passing extra fields if in autoprovisioning mode
+    """
+    if user:
+        return {'is_new': False}
+
+    fields = dict(
+        (name, kwargs.get(name) or details.get(name))
+        for name in (strategy.setting('USER_FIELDS', USER_FIELDS))
+    )
+
+    if not fields:
+        return
+
+    if not autoprovision:
+        return {
+            'is_new': True,
+            'user': strategy.create_user(**fields)
+        }
```

Comment:
@bradenmacdonald normal registration with autoprovisioning disabled *will* get to this point. It's not a fallback, it is basically the same code `social.pipeline.user.create_user` contains. I believe we're still supposed to keep non-autoprovisioned providers working, so this path is for them.